nix/shell.nix

```diff
@@ -0,0 +1,63 @@
+{ repoRoot, pkgs, ... }:
+
+cabalProject:
+
+{
```

Comment:
so... how do I change a tool version? Just override the derivation entirely?

Reply:
Yes, you'll have to provide your own derivation.

Reply:
If I want to change the version of a formatter, does that mean I'll also need to provide a HLS derivation that uses that version correctly?

Reply:
This is just making me want to immediately use our own derivations so this is easy to change later...

Reply:
Also... doesn't this mean that my version of iogx determines the formatter version? so updating iogx can cause me to have haskell formatting changes, which as I said before seems undesirable. Maybe other people don't care :shrug:

Reply:
iogx provides one toolchain per ghc, which includes hls, hlint, etc.
You can override any of those.
The version of iogx does not determine the formatter version, the GHC version does.

Reply:
>  The version of iogx does not determine the formatter version, the GHC version does.

So you promise to never ever change the version of a tool that's part of the toolchain for a given GHC? that seems unlikely - there will be bugs and you'll want to not recommend a broken version.